lib/drowsy/relation.rb

```diff
@@ -73,6 +73,18 @@ def to_http_request
     Drowsy::HttpRequest.new(connection, http_method, uri_template, params)
   end
 
+  def method_missing(name, *args, &block)
```

Comment:
Some thoughts, since you tagged me.

Reasons I tend to avoid method missing include:

- It doesn't give you as much visibility into the API via things like `public_methods`.
- Benchmarks are inherently slower than with `define_method`, etc.
- Sometimes encourages catch-all objects whose method_missing implementations handle multiple use cases at once.

Where I tend to use method_missing is in cases where defining the methods up-front would make the API surface massive, or where it really wouldn't be possible to know what the methods will be in advance (e.g. Her's use case with dynamic JSON attributes from an API).

As-is this will totally accomplish what you need, but keep in mind that since scopes are in fact finite and known in advance, there would be many other ways to actually define them as true methods on something.

Reply:
Yea. I have the same feelings about method missing, I just wanted to see if I could do this without defining the scope method on an instance of Relation which is another type of thing I generally try to avoid. I may reimplement now that I have the tests in there.

Reply:
Makes sense! 👍